File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:

        if not nums:
            return 0
        sortedNums = sorted(nums)

        curr = 1
        res = 1


        for i in range(len(sortedNums) - 1):
            if sortedNums[i] + 1 == sortedNums[i+1]:
                curr += 1
            elif sortedNums[i] != sortedNums[i+1]:
                res = max(res, curr)
                curr = 1

        return max(res, curr)




        

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        
        if (nums.empty())
        {
            return 0;
        }

        std::sort(nums.begin(), nums.end());

        for (int i = 1; i < nums.size(); i++)
        {
            if (nums[i] == nums[i - 1] + 1)
            {
                streak += 1;
            }
            else if (nums[i] == nums[i - 1])
            {
                //Duplicate
                continue;
            }
            else {
                streak = 1;
            }

            if (streak > res)
            {
                res = streak;
            }
        }

        return res;
    }

private:
    int res = 1;
    int streak = 1;
};